src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -78,10 +82,27 @@ public AuditUtility(
             }
         }
 
+        private void CheckPackageDownloadVulnerabilities(IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities)
+        {
+            foreach (var targetFramework in _targetFrameworks)
```

Comment:
Since `_targetFrameworks` is `IList`, this `foreach` loop will cause an allocation. And since audit runs in restore, which is our most perf-critical path, we need to do everything we can to minimize perf impacts.  So, either change this to a `for` loop and use the indexer to get the value from the list, or use `NoAllocEnumerate` (or whatever that extension method/class is called).